code/game/machinery/Sleeper.dm

```diff
@@ -306,6 +306,11 @@
 		return
 	if(!Adjacent(over_location))
 		return
+		
+	var/mob/user = usr
+	if (user.stunned || user.paralysis || user.resting) // If you're too disabled to put someone into it, you're too disabled to pull someone out of it.

```

Comment:
Look at mob_helpers.dm, there are procs to check this better. You probably want to use user.isUnconscious(), and check for user.restrained() while you're at it.